packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/builders/CategoryAwareNotificationBuilder.java

```diff
@@ -0,0 +1,73 @@
+package expo.modules.notifications.notifications.presentation.builders;
+
+import androidx.core.app.RemoteInput;
+import androidx.core.app.NotificationCompat;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.io.IOException;
+
+import expo.modules.notifications.notifications.model.NotificationAction;
+import expo.modules.notifications.notifications.model.NotificationCategory;
+import expo.modules.notifications.notifications.model.NotificationContent;
+import expo.modules.notifications.notifications.model.TextInputNotificationAction;
+import expo.modules.notifications.notifications.service.TextInputNotificationResponseReceiver;
+import expo.modules.notifications.notifications.service.SharedPreferencesNotificationCategoriesStore;
+import static expo.modules.notifications.notifications.service.NotificationResponseReceiver.getActionIntent;
+import static expo.modules.notifications.notifications.service.TextInputNotificationResponseReceiver.getActionIntent;
+
+public class CategoryAwareNotificationBuilder extends ExpoNotificationBuilder {
+  private SharedPreferencesNotificationCategoriesStore mStore;
+
+  public CategoryAwareNotificationBuilder(Context context) {
+    super(context);
+    mStore = new SharedPreferencesNotificationCategoriesStore(context);
```

Comment:
could we pass in the store instance in the constructor? (so it is possible to inject other class if needed)